File: SYP.Web/Modules/Catalog/Brands/BrandsForm.cs
using Serenity.ComponentModel;

namespace SYP.Catalog.Forms;

[FormScript("Catalog.Brands")]
[BasedOnRow(typeof(BrandsRow), CheckNames = true)]
public class BrandsForm
{
    public string Name { get; set; }

    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

    public string Logo { get; set; }

    [DefaultValue(1)]
    public short IsActive { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Catalog/BrandsForm.ts
import { ImageUploadEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";

export interface BrandsForm {
    Name: StringEditor;
    Description: TextAreaEditor;
    Logo: ImageUploadEditor;
    IsActive: IntegerEditor;
}

export class BrandsForm extends PrefixedContext {
    static readonly formKey = 'Catalog.Brands';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BrandsForm.init) {
            BrandsForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = ImageUploadEditor;
            var w3 = IntegerEditor;

            initFormType(BrandsForm, [
                'Name', w0,
                'Description', w1,
                'Logo', w2,
                'IsActive', w3
            ]);
        }
    }
}